### PR TITLE
KTOR-2763 Fix Warnings and Enable -Werror flag

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-2763](https://youtrack.jetbrains.com/issue/KTOR-2763)

### CODE REVIEW

File: ktor-client/ktor-client-tests/jvm/test/io/ktor/client/tests/utils/tests/CacheKtTest.kt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+package io.ktor.client.tests.utils.tests
+
+public class CacheKtTest {
```
Comment (rsinukov) [inline ktor-client/ktor-client-tests/jvm/test/io/ktor/client/tests/utils/tests/CacheKtTest.kt:3]: why empty test class?

**Review by rsinukov — APPROVED** [review]
lgtm, but minor comments

**Comment by rsinukov** [comment]
Please check CI build

File: build.gradle.kts [hunk 2]
```diff
@@ -159,11 +158,24 @@ allprojects {
...
+                }
+            }
+        }
+
         if (!disabledExplicitApiModeProjects.contains(project.name)) {
             explicitApi()
         }
 
-        sourceSets.matching { !(it.name in listOf("main", "test")) }.all {
+        sourceSets.matching {
```
[the remark this answers is not in the document]
Comment (e5l) [reply build.gradle.kts:176]: yep, let's fix

File: ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngine.kt [hunk 3]
```diff
@@ -19,6 +19,7 @@ import kotlinx.coroutines.*
 import kotlinx.coroutines.channels.*
 import kotlin.coroutines.*
 
+@OptIn(InternalAPI::class, kotlinx.coroutines.DelicateCoroutinesApi::class)
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-cio/common/src/io/ktor/client/engine/cio/CIOEngine.kt:22]: thanks, will fix

File: ktor-client/ktor-client-core/common/src/io/ktor/client/features/observer/DelegatedCall.kt [hunk 4]
```diff
@@ -53,6 +55,7 @@ internal class DelegatedRequest(
     origin: HttpRequest
 ) : HttpRequest by origin
 
+@InternalAPI
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/features/observer/DelegatedCall.kt:58]: should be OptIn instead, will fix

File: ktor-client/ktor-client-core/common/src/io/ktor/client/HttpClient.kt [hunk 5]
```diff
@@ -69,7 +69,7 @@ public fun HttpClient(
  * This is a generic implementation that uses a specific engine [HttpClientEngine].
  * @property engine: [HttpClientEngine] for executing requests.
  */
-@OptIn(InternalCoroutinesApi::class)
+@OptIn(InternalCoroutinesApi::class, io.ktor.util.InternalAPI::class)
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-core/common/src/io/ktor/client/HttpClient.kt:72]: fixed, thanks

File: ktor-client/ktor-client-features/ktor-client-auth/common/test/io/ktor/client/features/auth/DigestProviderTest.kt [hunk 6]
```diff
@@ -2,6 +2,8 @@
 * Copyright 2014-2021 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
 */
 
+@file:Suppress("DEPRECATION")
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-features/ktor-client-auth/common/test/io/ktor/client/features/auth/DigestProviderTest.kt:5]: sure

File: ktor-client/ktor-client-tests/jvm/test/io/ktor/client/tests/utils/tests/CacheKtTest.kt (hunk 1 above)
Comment (e5l) [reply ktor-client/ktor-client-tests/jvm/test/io/ktor/client/tests/utils/tests/CacheKtTest.kt:3]: generated accidentally, removed

File: ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/ClientLoaderJvm.kt [hunk 7]
```diff
@@ -43,19 +43,21 @@ public actual abstract class ClientLoader actual constructor(timeoutSeconds: Int
...
             val platformShouldBeSkipped = "*" == platform || OS_NAME == platform
-            val engineShouldBeSkipped = "*" == engine || this.engine.toString().toLowerCase() == engine
+            val engineShouldBeSkipped = "*" == engine || this.engine.toString().lowercase(Locale.getDefault()) == engine
 
             if (platformShouldBeSkipped && engineShouldBeSkipped) {
                 return
             }
         }
-        if (skipEngines.map { it.toLowerCase() }.contains(engine.toString().toLowerCase())) return
+        if (skipEngines.map { it.lowercase(Locale.getDefault()) }.contains(
```
[the remark this answers is not in the document]
Comment (e5l) [reply ktor-client/ktor-client-tests/jvm/src/io/ktor/client/tests/utils/ClientLoaderJvm.kt:52]: done

**Pull request merged**